File: 0206-Reverse-Linked-List/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        temp = head
        temp_list = []
        end = -1
        while(temp != None):
            temp_list.append(temp.val)
            temp = temp.next
        temp = head
        for i in range(len(temp_list)):
            temp.val = temp_list[end - i]
            temp = temp.next
        return head

File: 0206-Reverse-Linked-List/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if(head == null){
            return head;
        }
        if(head.next == null){
            return head;
        }
        ListNode newHead = reverseListHelper(head);
        head.next = null;
        return newHead;
    }

    ListNode reverseListHelper(ListNode temp){
        if(temp.next == null){
            return temp;
        }
        ListNode tempNext = temp.next;
        ListNode newHead = reverseListHelper(tempNext);
        tempNext.next = temp;
        return newHead;

    }
}